### ISSUE TITLE
'Unexpected end of JSON input' with 'gatsby-transformer-remark' preventing Netlify build

### ISSUE THREAD

**Issue by sunny-kim-92** [gatsbyjs/gatsby#18147]
## Description

I've been running into a number of errors when trying to introduce plugins for adding images directly in my markdown. I have since removed instances of images within markdown, but an error remains and it seems to be preventing Netlify from building. 

I've posted the issue on Netlify as well because `gatsby build` and `gatsby serve` both work as intended. So this may be directed at the wrong place, I apologize. But after reading through a number of issues related to 'gatsby-sharp', I wanted to cover my bases.

I attempted to use gatsby-remark-images and add the following to my markdown:

`
![some caption text](../.path/to/image)
`

I would get the following error on `gatsby develop`:
```
objc[6387]: Class GNotificationCenterDelegate is implemented in both /Users/home/Code/amici-site/node_modules/sharp/vendor/lib/libgio-2.0.0.dylib (0x10768d578) and /Users/home/Code/amici-site/node_modules/gatsby-transformer-sharp/node_modules/sharp/vendor/lib/libgio-2.0.0.dylib (0x119ca3578). One of the two will be used. Which one is
```

And this error on `gatsby develop` and Netlify build logs:
```

 ERROR #11321  PLUGIN

"gatsby-transformer-json" threw an error while running the onCreateNode lifecycle:

Unexpected end of JSON input



  SyntaxError: Unexpected end of JSON input

  - JSON.parse

  - gatsby-node.js:63 Object.onCreateNode
    [amici-site]/[gatsby-transformer-json]/gatsby-node.js:63:30

```

After reading through some of the GitHub issues (I can't seem to find the link), I ended up downgrading gatsby-transformer-sharp to 2.2.14, which fixed the second error on `gatsby develop`, but it still appeared in the Netlify build logs. I have since removed the image syntax from my markdown, but the first error still remains in Netlify. I apologize again if this is in the wrong place.

### Steps to reproduce

https://github.com/sunny-kim-92/amici-site. One thing I noticed, this may be way out of left fied, but in my gatsby-config.js, if I change one pair of single quotes to double quotes and run `gatsby develop`, I get the 'JSON Parse' error. But when I run it a second time, keeping the same double quotes, the error goes away.

### Expected result

Page works as anticipated upon `gatsby develop`, even with image in markdown, but the error appears and Netlify will not complete a build.

### Actual result

What happened.

### Environment

```
  System:
    OS: macOS 10.14.4
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.0 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 77.0.3865.90
    Firefox: 69.0.1
    Safari: 12.1
  npmPackages:
    gatsby: ^2.15.29 => 2.15.29
    gatsby-image: ^2.2.24 => 2.2.24
    gatsby-mdx: ^1.0.0 => 1.0.0
    gatsby-plugin-eslint: ^2.0.5 => 2.0.5
    gatsby-plugin-mdx: ^1.0.47 => 1.0.47
    gatsby-plugin-nprogress: ^2.1.9 => 2.1.9
    gatsby-plugin-offline: ^3.0.11 => 3.0.11
    gatsby-plugin-react-helmet: ^3.1.10 => 3.1.10
    gatsby-plugin-react-svg: ^2.1.2 => 2.1.2
    gatsby-plugin-sharp: ^2.2.28 => 2.2.28
    gatsby-plugin-sitemap: ^2.2.16 => 2.2.16
    gatsby-plugin-styled-components: ^3.1.8 => 3.1.8
    gatsby-plugin-webpack-size: ^1.0.0 => 1.0.0
    gatsby-remark-component: ^1.1.3 => 1.1.3
    gatsby-remark-external-links: 0.0.4 => 0.0.4
    gatsby-remark-figure-caption: ^2.0.0 => 2.0.0
    gatsby-remark-images: ^3.1.25 => 3.1.25
    gatsby-source-filesystem: ^2.1.29 => 2.1.29
    gatsby-transformer-json: ^2.2.12 => 2.2.12
    gatsby-transformer-remark: ^2.6.27 => 2.6.27
    gatsby-transformer-sharp: ^2.2.20 => 2.2.20
  npmGlobalPackages:
    gatsby-cli: 2.7.26
```

**Comment by darthzyklus** [comment]
@sunny-kim-92 did you fix this? what did you do?

**Comment by sunny-kim-92** [comment]
@darthzyklus So I did, but it's kind of an incomplete answer. There was an empty .json file nested really deep in the folder that the 'gatsby-transformer-remark' plugin was pointing to. When I removed the file, the 'end of JSON' error disappeared and the Netlify build worked.

But if you were talking about 'gatsby-transformer-sharp' plugin, I only changed the pictures I imported into Markdown, and the error stopped for some reason. I didn't change anything in gatsby-config, and both pictures were .png. The only GitHub issue I could find that really referenced the warning and had a conversation is here: https://github.com/CocoaPods/CocoaPods/issues/7155. But it's for CocoaPods...I actually recently started working with deployment on xCode on my machine, but it had nothing to do with Gatsby/Node, so I feel it's (extremely) unlikely it relates. There was an issue that didn't reference the same error but referenced 'gatsby-transformer-sharp', which recommended downgrading 'gatsby-transformer-sharp' to 2.2.14, I can't find the issue for some reason. I hope any of that helps.
